src/node/nodestate.h

```diff
@@ -255,64 +255,114 @@ namespace ccf
     std::vector<uint8_t> serialize_create_request(
       const CreateNew::In& args, const std::vector<uint8_t>& quote)
     {
-      jsonrpc::ProcedureCall<CreateNetworkNodeToNode::In> create_rpc;
-
-      create_rpc.id = 0;
-      create_rpc.method = ccf::MemberProcs::CREATE;
+      CreateNetworkNodeToNode::In create_params;
 
       for (auto& m_info : args.config.genesis.members_info)
       {
-        create_rpc.params.members_info.push_back(m_info);
+        create_params.members_info.push_back(m_info);
       }
 
-      create_rpc.params.gov_script = args.config.genesis.gov_script;
-      create_rpc.params.node_cert = node_cert;
-      create_rpc.params.network_cert = network.identity->cert;
-      create_rpc.params.quote = quote;
-      create_rpc.params.public_encryption_key =
+      create_params.gov_script = args.config.genesis.gov_script;
+      create_params.node_cert = node_cert;
+      create_params.network_cert = network.identity->cert;
+      create_params.quote = quote;
+      create_params.public_encryption_key =
         node_encrypt_kp->public_key_pem().raw();
-      create_rpc.params.code_digest =
+      create_params.code_digest =
         std::vector<uint8_t>(std::begin(node_code_id), std::end(node_code_id));
-      create_rpc.params.node_info_network = args.config.node_info_network;
+      create_params.node_info_network = args.config.node_info_network;
 
-      nlohmann::json j = create_rpc;
-      auto contents = nlohmann::json::to_msgpack(j);
+      const auto body = jsonrpc::pack(create_params, jsonrpc::Pack::Text);
 
-      auto sig_contents = node_sign_kp->sign(contents);
+      http::Request request(
+        fmt::format("/{}/{}", ccf::Actors::MEMBERS, ccf::MemberProcs::CREATE));
+      request.set_header(
+        http::headers::CONTENT_TYPE, http::headervalues::contenttype::JSON);
 
-      nlohmann::json sj;
-      sj["req"] = j;
-      sj["sig"] = sig_contents;
+      request.set_body(&body);
+      http::sign_request(request, node_sign_kp);
 
-      return jsonrpc::pack(sj, jsonrpc::Pack::Text);
+      return request.build_request();
     }
 
-    void send_create_request(const std::vector<uint8_t>& packed)
+    bool parse_create_response(const std::vector<uint8_t>& response)
     {
-      constexpr auto actor = ccf::ActorsType::members;
+      http::SimpleMsgProcessor processor;
+      http::Parser parser(HTTP_RESPONSE, processor);
 
-      auto handler = this->rpc_map->find(actor);
-      if (!handler.has_value())
+      const auto parsed_count =
+        parser.execute(response.data(), response.size());
+      if (parsed_count != response.size())
       {
-        throw std::logic_error("Handler has no value");
+        LOG_FAIL_FMT(
```

Comment:
I added these while debugging an issue with this special create request. These logging lines don't actually work, because the host isn't processing outbound messages from the enclave yet. I'm not sure what the right solution is. Maybe we pump outbound messages even if enclave create fails? Maybe all of this create processing should be moved to a separate file with its restricted environment clearly documented?

Reply:
How does this happen so early? This is parsing an HTTP request, the enclave should already be setup by then? Pumping messages even on failure seems like a good idea in any case.

Reply:
This is parsing the `Create` request, which is locally created - not a user RPC. I believe we made it look like an RPC so it can go through the user frontend and be committed+replicated as normal, @jumaffre may understand this in more detail?

Reply:
This was added for PBFT so that the first transaction is also a request (executed by the member frontend), but is also applicable for Raft of course. I thought the host would eventually process messages that have been written to the ring buffer by the enclave but it seems that it's not the case.

Reply:
Ah yes, I remember now. I think we do need to have the host read the outbound RB even on startup failure then.

Reply:
As discussed, looks like the right way to resolve this is just to remove the `create` ECALL and do all this work within `enclave_run`: https://github.com/microsoft/CCF/issues/857